### PR TITLE
fix: should have counter for each action

### PR DESCRIPTION
**Pull request by hreflee**
Proposing a fix for https://github.com/bhovhannes/redux-profiler/issues/651

### CODE REVIEW

**Comment by bhovhannes** [comment]
I think I had github workflows configured wrong, that's why `lint` and `test` checks do not run.
@hreflee please merge the latest `main` branch into yours to let them run.

**Comment by hreflee** [comment]
@bhovhannes I have pulled the latest `main` branch

**Comment by bhovhannes** [comment]
> @bhovhannes I have pulled the latest `main` branch

Good, all checks passed. I've left 2 comments on PR. Please have a look and PR is good to go.

**Comment by hreflee** [comment]
> > @bhovhannes I have pulled the latest `main` branch
> 
> Good, all checks passed. I've left 2 comments on PR. Please have a look and PR is good to go.

Some reformatting changes are made by lint in commit hook. I am also confused about these changes. Does I need to remove them?

**Comment by bhovhannes** [comment]
> Some reformatting changes are made by lint in commit hook. I am also confused about these changes. Does I need to remove them?

No, that's fine.

**Comment by hreflee** [comment]
> > @bhovhannes I have pulled the latest `main` branch
> 
> Good, all checks passed. I've left 2 comments on PR. Please have a look and PR is good to go.

I do not see your comments. I just see 2 comments from Codecov. Does you mean the 2 comments from Codecov?

File: src/index.js [hunk 1]
```diff
@@ -75,7 +68,11 @@ export default function profileStore(options = {}) {
 			performance.clearMarks(markName + START)
 			performance.clearMarks(markName + END)
 			performance.clearMeasures(getMarkLabel(markInfo, markType))
-			counter -= counterDelta
+			if (counter[markInfo.name]) {
+				counter[markInfo.name]--
+			} else {
+				delete counter[markInfo.name]
+			}
```
Comment (bhovhannes) [inline src/index.js:75]: I think there is a bug here, property will be deleted only in the next run. Shouldn't this be:
```suggestion
			counter[markInfo.name]--
			if (!counter[markInfo.name]) {
				delete counter[markInfo.name]
			}
```

File: src/index.js [hunk 2]
```diff
@@ -75,7 +68,10 @@ export default function profileStore(options = {}) {
 			performance.clearMarks(markName + START)
 			performance.clearMarks(markName + END)
 			performance.clearMeasures(getMarkLabel(markInfo, markType))
-			counter -= counterDelta
+			counter[markInfo.name]--
+			if (!counter[markInfo.name]) {
```
Comment (bhovhannes) [inline src/index.js:72]: Left a bug in my previous suggestion. This should fix it:
```suggestion
			if (counter[markInfo.name] <= 0) {
```

File: src/index.js [hunk 3]
```diff
@@ -75,7 +68,10 @@ export default function profileStore(options = {}) {
 			performance.clearMarks(markName + START)
 			performance.clearMarks(markName + END)
 			performance.clearMeasures(getMarkLabel(markInfo, markType))
-			counter -= counterDelta
+			counter[markInfo.name]--
+			if (counter[markInfo.name] <= 0) {
+				delete counter[markInfo.name]
+			}
```
Comment (bhovhannes) [inline src/index.js:74]: The more I think the more I like your version. Reverting to it to merge the MR. Sorry for confusion.
```suggestion
			if (counter[markInfo.name]) {
				counter[markInfo.name]--
			} else {
				delete counter[markInfo.name]
			}
```

**Comment by bhovhannes** [comment]
this fixes #651

**Pull request merged**